Lib9c/Policy/BlockPolicySource.cs

```diff
@@ -338,22 +332,61 @@ public partial class BlockPolicySource
                     // FIXME: This works under a strong assumption that any miner that was ever
                     // in a set of authorized miners can only create transactions without
                     // any actions.
-                    return transaction.Actions.Any()
+                    return ((ITransaction)transaction).CustomActions?.Any() is true
                         ? new TxPolicyViolationException(
                             $"Transaction {transaction.Id} by an authorized miner should not " +
-                            $"have any action: {transaction.Actions.Count}",
+                            $"have any action: {((ITransaction)transaction).CustomActions?.Count}",
                             transaction.Id)
                         : null;
                 }
 
                 // Check ActivateAccount
-                if (transaction.CustomActions is { } customActions &&
+                if (((ITransaction)transaction).CustomActions is { } customActions &&
```

Comment:
Can we load an action type dynamically, based on `blockChain.Tip` here too?

Reply:
I tried to apply it in [43993cb](https://github.com/planetarium/lib9c/pull/1580/commits/43993cb9eacb320f4debf57ac99e22d1a89f3478), [283851f](https://github.com/planetarium/lib9c/pull/1580/commits/283851f54010065919676680221bb24838c71e61), [1052598](https://github.com/planetarium/lib9c/pull/1580/commits/1052598f2fa6c5ce14df85addb40dc8490b63fe6)